### DISCUSSION TITLE
Using encode/databases

### DISCUSSION THREAD

**Discussion by jocho-here** [fastapi/fastapi#4369] [Questions] [answered]
Hi,

I recently found out about `encode/databases` library, and I'm giving it a try.
While testing it on the cloud, I started seeing:
```
asyncpg.exceptions.TooManyConnectionsError: too many connections for role "vslyvkeneoptxy"
```

This is how I use it:
- I'm initializing the global `database` object in the startup, just like how it's shown in its documentation.
- Once it's initialized, I'm letting all the `APIRouter` use this global object and do `fetch...` or `execute` on raw queries.

I'm assuming it takes care of the pool and connections by itself. But why the too many connections issue though..?

Has anyone given `databases` library a try?

**Comment by Kludex** [comment, marked as answer]
I recommend using [SQLAlchemy](https://www.sqlalchemy.org/) or [SQLModel](https://sqlmodel.tiangolo.com/) instead.
